types/underscore/index.d.ts

```diff
@@ -95,18 +101,15 @@ declare module _ {
         (element: T, key: string, object: V): TResult;
     }
 
-    type EnumerableKey = string | number;
-
-    type CollectionIterator<T, TResult, V> =
-        V extends List<T> ? ListIterator<T, TResult, V>
-        : V extends Dictionary<T> ? ObjectIterator<T, TResult, V>
-        : never;
+    interface CollectionIterator<T, TResult, V> {
```

Comment:
Now that a lot of functions in this library are becoming value-centric and `ListIterator` and `ObjectIterator` will likely be completely deprecated like `MemoIterator` and `MemoObjectIterator` already are, I wonder if it would be worth making `CollectionIterator` and `MemoCollectionIterator` value-centric as well (i.e. making the former `CollectionIterator<V, TResult>` and using `TypeOfCollection<V>` to get what is currently `T`?

I guess since `element` is the first parameter one could potentially want to make one of these without caring what `V` is, so maybe I should leave it mostly as-is but default `V` to `Collection<T>` or something to allow it to be used that way instead.

Reply:
To be honest I don't really care about type parameters in general, since I'm firmly in the school of wanting to avoid type annotations whenever possible. In that regard, you can change them or keep them as they are and I'll be happy either way.

That said, iterator functions that list only a value parameter are definitely a thing.

Reply:
I've given this a decent amount of thought and decided that I like the current ordering. Thanks!